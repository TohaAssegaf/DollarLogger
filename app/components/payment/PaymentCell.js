import styles from './styles'
import * as MoneyFormatter from '~/app/lib/MoneyFormatter'
import * as DateUtils from '~/app/lib/DateUtils'
import React from 'react'
import { Text, TouchableHighlight, View } from 'react-native'

export default class PaymentCell extends React.Component {
  render() {
    return (
      <TouchableHighlight onPress={() => this.props.onTap()} underlayColor='#EEE'>
        <View style={styles.paymentCell}>
          <View style={styles.paymentCellDate}>
            <Text style={styles.paymentCellText}>
              {DateUtils.getShortFormat(this.props.payment.date)}
            </Text>
          </View>
          <View style={styles.paymentCellName}>
            <Text style={styles.paymentCellText} numberOfLines={2}>
              {this.props.payment.name}
            </Text>
          </View>
          <View style={styles.paymentCellMoney}>
            <Text style={styles.paymentCellText}>
              {MoneyFormatter.formatMoney(this.props.payment.total)}
            </Text>
          </View>
        </View>
      </TouchableHighlight>
    );
  }
}
