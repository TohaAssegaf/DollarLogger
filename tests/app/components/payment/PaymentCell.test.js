import PaymentCell from '~/app/components/payment/PaymentCell'
import React from 'react';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const rendered = renderer.create(<PaymentCell payment={{
    name: "Test payment",
    total: 100,
    date: new Date(2018, 4, 2)}}
  />).toJSON();
  expect(rendered).toMatchSnapshot();
});
