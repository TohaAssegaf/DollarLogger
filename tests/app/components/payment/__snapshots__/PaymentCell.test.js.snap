// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={null}
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderBottomColor": "#222222",
        "borderBottomWidth": 0.5,
        "flexDirection": "row",
        "height": 60,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flex": 0.15,
          "paddingHorizontal": 20,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 16,
          }
        }
      >
        5/2
      </Text>
    </View>
    <View
      style={
        Object {
          "flex": 0.6,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={2}
        style={
          Object {
            "fontSize": 16,
          }
        }
      >
        Test payment
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "flex": 0.25,
          "paddingRight": 20,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 16,
          }
        }
      >
        $1.00
      </Text>
    </View>
  </View>
</View>
`;
